File: 9012.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
	int T;
	cin >> T;
	cin.ignore();

	while(T--) {
		string line;
		getline(cin, line);
		stack<char> pstack;
		string is_VPS = "YES";
		
		for (char ps : line) {
			if (ps == '(') {
				pstack.push(ps);
			}
			else {
				if (!pstack.empty()) {
					pstack.pop();
				}
				else {
					// ')'가 남아있는데 pstack이 비어있을 경우
					is_VPS = "NO";
					break;
				}
			}
		}
		
		if (!pstack.empty()) {
			is_VPS = "NO";
		}
		cout << is_VPS << endl;
	}

	return 0;
}

File: 9012.py
import sys

N = int(sys.stdin.readline())
for _ in range(N):
    input = list(sys.stdin.readline().rstrip())
    pstack=[]
    is_VPS="YES"
    for ps in input:
        if ps == "(":
            pstack.append(ps)
        else:
            if not pstack:
                # ')' 가 남아있는데 pstack이 비어있을 경우
                is_VPS = "NO"
                break
            else:
                pstack.pop()
    
    if pstack:
        is_VPS = "NO"
        
    print(is_VPS)